tests/test_polling_manager.py

```diff
@@ -35,3 +36,21 @@ def test_polling_manager_calls_update_environment_on_each_refresh():
     # for each subsequent refresh
     assert flagsmith.update_environment.call_count == 3
     polling_manager.stop()
+
+
+def test_polling_manager_is_resilient_to_api_errors(server_api_key):
+    with mock.patch("requests.Session") as session_get_mock:
+        # Given
+        session_get_mock.return_value = mock.MagicMock(
+            get=mock.MagicMock(status_code=500)
+        )
```

Comment:
I'd be interested to know if it's also resilient to `requests.RequestException`, not just 500 status code responses. Could we add another test case for that?

Reply:
No, it's not resilient to `RequestException` only [`FlagsmithAPIError`](https://github.com/Flagsmith/flagsmith-python-client/pull/60/files#diff-a729df2f71a8cb45578b203868ef70fa16ab97dcbb092331909e20f5afafe646R31), which in turn is raised in case of [`JSONDecodeError` and `JSONDecodeError`](https://github.com/Flagsmith/flagsmith-python-client/blob/d334564779f464f5625a90d93611faee56fce1cc/flagsmith/flagsmith.py#L271).

Reply:
Added in 022a9fd31c470c425349101f94d7b9a4e099e28e